activation/handler.go

```diff
@@ -172,27 +172,58 @@ func (h *Handler) syntacticallyValidate(ctx context.Context, atx *wire.Activatio
 	return nil
 }
 
+func (h *Handler) commitment(ctx context.Context, atx *wire.ActivationTxV1) (types.ATXID, error) {
+	if atx.PrevATXID == types.EmptyATXID {
+		return *atx.CommitmentATXID, nil
+	}
+	return atxs.CommitmentATX(h.cdb, atx.SmesherID)
+}
+
+func (h *Handler) previous(ctx context.Context, id types.ATXID) (*types.ActivationTx, error) {
+	var blob sql.Blob
+	if err := atxs.LoadBlob(ctx, h.cdb, id[:], &blob); err != nil {
+		return nil, err
+	}
+
+	if blob.Bytes == nil {
+		// if the blob is empty, it means the ATX is golden - fetch it from the DB
+		vatx, err := atxs.Get(h.cdb, id)
+		if err != nil {
+			return nil, fmt.Errorf("fetching golden previous atx: %w", err)
+		}
+		return vatx.ActivationTx, nil
+	}
+
+	var atx wire.ActivationTxV1
+	if err := codec.Decode(blob.Bytes, &atx); err != nil {
+		return nil, fmt.Errorf("decoding previous atx: %w", err)
+	}
+	return wire.ActivationTxFromWireV1(&atx, blob.Bytes...), nil
```

Comment:
I would return the wire type here to make the difference between domain and wire type more clear. We need both ATXs in wire form to be able to create the domain time for the current ATX further down the validation path.